stubs/pexpect/pexpect/spawnbase.pyi

```diff
@@ -16,66 +21,96 @@ class _Logfile(Protocol):
     def write(self, __s) -> object: ...
     def flush(self) -> object: ...
 
-class SpawnBase:
-    encoding: Incomplete
-    pid: Incomplete
+_InputStringPattern: TypeAlias = str | bytes | EOF | TIMEOUT
+_InputRePattern: TypeAlias = Pattern[str] | Pattern[bytes] | _InputStringPattern
+_CompiledStringPattern: TypeAlias = AnyStr | EOF | TIMEOUT
+_CompiledRePattern: TypeAlias = Pattern[AnyStr] | EOF | TIMEOUT
+_Searcher: TypeAlias = searcher_string[AnyStr] | searcher_re[AnyStr]
+
+class SpawnBase(Generic[AnyStr]):
+    encoding: str | None
+    pid: int | None
     flag_eof: bool
-    stdin: Incomplete
-    stdout: Incomplete
-    stderr: Incomplete
-    searcher: Incomplete
+    stdin: TextIO
+    stdout: TextIO
+    stderr: TextIO
+    searcher: None
     ignorecase: bool
-    before: Incomplete
-    after: Incomplete
-    match: Incomplete
-    match_index: Incomplete
+    before: AnyStr | None
+    after: _CompiledStringPattern[AnyStr] | None
+    match: AnyStr | Match[AnyStr] | EOF | TIMEOUT | None
+    match_index: int | None
     terminated: bool
-    exitstatus: Incomplete
-    signalstatus: Incomplete
-    status: Incomplete
+    exitstatus: int | None
+    signalstatus: int | None
+    status: int | None
     child_fd: int
-    timeout: Incomplete
-    delimiter: Incomplete
+    timeout: float | None
+    delimiter: type[EOF]
     logfile: _Logfile
```

Comment:
While you're making changes here, could you change `logfile` and friends to `_Logfile | None`?

Reply:
fixed